### PR TITLE
coursier: Add support for both `cs` and `coursier` executable names

### PR DESCRIPTION
**Pull request by Holzhaus**
On some systems, the executable might be named `coursier` instead of
`cs`. For example, this is the case on Arch Linux when using the AUR
package, or when following the official instructions when installing the
JAR-based launcher:

  https://get-coursier.io/docs/cli-installation#jar-based-launcher

### CODE REVIEW

File: pre_commit/languages/coursier.py [hunk 1]
```diff
@@ -27,6 +27,14 @@ def install_environment(
     helpers.assert_version_default('coursier', version)
     helpers.assert_no_additional_deps('coursier', additional_dependencies)
 
+    # Support both possible executable names (either "cs" or "coursier")
+    executable = next(filter(helpers.exe_exists, ('cs', 'coursier')), None)
```
Comment (asottile) [inline pre_commit/languages/coursier.py:31]: unnecessary FP -- we avoid `map` / `filter` / `reduce` 

this should be `exe('cs') or exe('coursier')` once you find the right function (`helpers.exe_exists` isn't it -- that returns a `bool`)

Comment (Holzhaus) [reply pre_commit/languages/coursier.py:31]: > this should be `exe('cs') or exe('coursier')` once you find the right function (`helpers.exe_exists` isn't it -- that returns a `bool`)

Do you want me to write a new one or is there something that I can use? `shutil.which` would be a candiate, or maybe `pre_commit.parse_shebang.find_executable` which kind of duplicates that functionality.

Comment (asottile) [reply pre_commit/languages/coursier.py:31]: `find_executable` is the right one probably yes

**Review by asottile — APPROVED** [review]
thanks!

**Pull request merged**